vms/platformvm/block/executor/manager.go

```diff
@@ -127,12 +127,34 @@ func (m *manager) VerifyTx(tx *txs.Tx) error {
 		return ErrChainNotSynced
 	}
 
-	return tx.Unsigned.Visit(&executor.MempoolTxVerifier{
-		Backend:       m.txExecutorBackend,
-		ParentID:      m.preferred,
-		StateVersions: m,
-		Tx:            tx,
+	stateDiff, err := state.NewDiff(m.preferred, m)
```

Comment:
I think we already use `VerifyTx` exclusively for mempool verification (including in the `network` pkg). Since we're not modifying the function usage, I think the same name is okay and keeps the diff minimal